instrumentation-core/spring/README.md

```diff
@@ -340,6 +341,274 @@ After running Jaeger locally, navigate to the url below. Make sure to refresh th
+***Note: The default port for the Apache Tomcat is 8080. On localhost both FirstService and SecondService services will attempt to run on this port raising an error. To avoid this add `server.port=8081` to the resources/application.properties file. Ensure the port specified corresponds to port referenced by FirstServiceController.SECOND_SERVICE_URL. ***
 
 Congrats, we just created a distributed service with OpenTelemetry!
+
+## Manual Instrumentation using Handlers and Filters
+
+In this section, we will implement the javax Serverlet Filter interface to wrap all requests to FirstService and SecondService controllers in a span. 
+
+We will also use the RestTemplate HTTP client to send requests from FirstService to SecondService. To propagate the trace in this request we will also implement the ClientHttpRequestInterceptor interface. This implementation is only required for projects that send outbound requests. In this example it is only required for FirstService. 
+
+### Set up FirstService and SecondService
+
+Using the earlier instructions [create two spring projects](#create-two-spring-projects) and add the required [dependencies and configurations](#setup-for-manual-instrumentation). 
+
+### Instrumentation of Client Service: SecondService
+
+Ensure the main method in SecondServiceApplication is defined. This will be the entry point to the SecondService project. This file should be created by the Spring Boot project initializer.
+
+```java
+import java.io.IOException;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class SecondServiceApplication {
+
+   public static void main(String[] args) throws IOException {
+      SpringApplication.run(SecondServiceApplication.class, args);
+   }
+}
+```
+
+Add the REST controller below to your SecondService project. This controller will return a string when SecondServiceController.secondTracedMethod is called:
+
+```java
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/time")
+public class SecondServiceController {
+   @Autowired
+   private Tracer tracer;
+
+   @GetMapping
+   public String secondTracedMethod() {
+      return "It's time to get a watch";
+   }
+}
+```
+
+#### Create Controller Filter
+
+Add the class below to wrap all requests to the SecondServiceController in a span. This class will call the preHandle method before the REST controller is entered and the postHandle method after a response is created. 
+
+The preHandle method starts a span for each request. This implementation is shown below:    
+
+```java
+
+@Component
+public class ControllerFilter implements Filter {
```

Comment:
i forget, does this get automatically picked up by Spring, do we need to define a url mapping for it?